MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+        NSEntityDescription *newDropDescription = [NSEntityDescription entityForName:@"AreaDrop" inManagedObjectContext:self.managedObjectContext];
+        drop = [[AreaDrop alloc] initWithEntity:newDropDescription insertIntoManagedObjectContext:self.managedObjectContext];
+        drop.idDecimalString = idDecimalString;
+        drop.method = areaDropDict[@"method"];
+        NSString *percent = areaDropDict[@"chance"];
+        drop.percentValue = percent.integerValue;
+        drop.quantity = areaDropDict[@"number"];
+        drop.rank = areaDropDict[@"rank"];
+        
+        // Hook up the area relationship.  We already ensured the area exists.
+        drop.area = area;
+        [area.dropSet addObject:drop];
+        
+        // Time to hook up the item relationship.  The item may or may not already exist.
+        NSString *itemName = areaDropDict[@"item"];
+        Item *item = (Item *)[self getUniqueEntityWithEntityName:@"Item" withNameAttribute:itemName];
+        if (!item) {
+            NSEntityDescription *newItemDescription = [NSEntityDescription entityForName:@"Item" inManagedObjectContext:self.managedObjectContext];
+            item = [[Item alloc] initWithEntity:newItemDescription insertIntoManagedObjectContext:self.managedObjectContext];
+            item.name = itemName;
+        }
+        [item.areaSourceSet addObject:drop];
+        drop.item = item;
+        
+        // The Drop should now exist, and be properly hooked up to the area, which exists and is hooked up to the region.
+    }
+}
+
+# pragma mark - Helper Methods
+
+- (NSDictionary *)loadDictionaryFromJsonFileNamed:(NSString *)fileName
+{
+    NSString *filePath = [[NSBundle mainBundle] pathForResource:fileName ofType:@"json"];
+    NSError *error;
+    NSString *fileContents = [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:&error];
+    if (error) {
+        NSLog(@"Error reading file: %@", error.localizedDescription);
+        return nil;
+    }
+    return (NSDictionary *)[NSJSONSerialization JSONObjectWithData:[fileContents dataUsingEncoding:NSUTF8StringEncoding]
+                                                           options:0
+                                                             error:nil];
+}
+
+- (NSArray *)loadArrayFromJsonFileNamed:(NSString *)fileName
+{
+    NSString *filePath = [[NSBundle mainBundle] pathForResource:fileName ofType:@"json"];
+    NSError *error;
+    NSString *fileContents = [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:&error];
+    if (error) {
+        NSLog(@"Error reading file: %@", error.localizedDescription);
+        return nil;
+    }
+    return (NSArray *)[NSJSONSerialization JSONObjectWithData:[fileContents dataUsingEncoding:NSUTF8StringEncoding]
+                                                      options:0
+                                                        error:nil];
+}
+
+- (BOOL)didSuccessfullySaveContext
```

Comment:
This is misleadingly named—it sounds like a (delegate) method that gets called after the save, but it's actually attempting the save.